### ISSUE TITLE
Python chunk output to console

### ISSUE THREAD

**Issue by rfenouil** [rstudio/reticulate#800]
Hello,

I have been using rmarkdown to generate html reports with R, and I now try to integrate some python chunks too.

Usually, I use the R function `message()` to output progress in the console when knitting. 
I believed that `stderr` messages are sent to the console while `stdout` ones are captured for knitting.

While trying to do the same with python, I think that both `stdout` and `stderr` streams are captured for knitting, and I find no way to print my usual progress messages in the console.

My typical R chunk:
```
{r testoutput_r}
# stdout
cat("This goes to HTML output !")

# stderr
message("This goes to console !")
```

Trying to do the same with python:
```
{python testoutput_python}
import sys

# stdout
print("\nThis goes to HTML output !")

# stderr
print("\nAnd this goes to HTML too...", file=sys.stderr, flush=True)

# stderr again
sys.stderr.write("\nAnd HTML again..." )
sys.stderr.flush()
```

Could you help me to control where to send my text output ?
Thank you.

```
Python version 3.7.3 (default, Dec 20 2019, 18:57:59) [GCC 8.3.0]
```

```
R version 3.6.3 (2020-02-29)

Platform: x86_64-pc-linux-gnu (64-bit)

locale: LC_CTYPE=en_US.UTF-8, LC_NUMERIC=C, LC_TIME=en_US.UTF-8, LC_COLLATE=en_US.UTF-8, LC_MONETARY=en_US.UTF-8, LC_MESSAGES=C, LC_PAPER=en_US.UTF-8, LC_NAME=C, LC_ADDRESS=C, LC_TELEPHONE=C, LC_MEASUREMENT=en_US.UTF-8 and LC_IDENTIFICATION=C
```

**Comment by kevinushey** [comment]
I might be missing something, but here's what I see with this document:

`````
---
title: "example"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r testoutput_r}
# stdout
cat("This goes to HTML output !")

# stderr
message("This goes to console !")
```

```{python testoutput_python}
import sys

# stdout
print("\nThis goes to HTML output !")

# stderr
print("\nAnd this goes to HTML too...", file=sys.stderr, flush=True)

# stderr again
sys.stderr.write("\nAnd HTML again..." )
sys.stderr.flush()
```
`````

<img width="707" alt="Screen Shot 2020-07-21 at 9 28 49 AM" src="https://user-images.githubusercontent.com/1976582/88080928-96fb2200-cb34-11ea-9405-30778a1fde4d.png">

In other words, in both cases both stdout and stderr is captured in the generated document.

**Comment by rfenouil** [comment]
Hello and thank you for your help.
You are correct I did not give you enough elements to reproduce, sorry.

I am used to set chunk option `message = FALSE` to get this behavior.

```
knitr::opts_chunk$set(echo = TRUE, message = FALSE)
```

BTW, Is it considered bad practice ?

**Comment by kevinushey** [comment]
Thanks! I think ultimately then the issue here is that the `reticulate` engine doesn't properly handle the `message` chunk option.

I don't think it's considered bad practice; it's just something that's not supported by `reticulate` in its own chunks yet.

**Comment by rfenouil** [comment]
Ok, thank you for your feedback I understand better now.

Do you know if it's something that you want to implement in a near future ? 
I realize it is probably not a critical feature though. I'll let you decide if you want to close the issue or keep it open.

Thanks again for the great work with reticulate !

**Comment by kevinushey** [comment]
This is something we should implement eventually; I'm just not sure when it will happen. A pull request would be appreciated :-)

**Comment by rfenouil** [comment]
I wish I could help but it's unfortunately far beyond my skills... 
Once I'll feel comfortable enough with python and reticulate I'll come and check the status here to see if I can still be useful :)
